jsonpb/jsonpb.go

```diff
@@ -73,8 +73,40 @@ type Marshaler struct {
 
 	// Whether to use the original (.proto) name for fields.
 	OrigName bool
+
+	// A custom URL resolver to use when marshaling Any messages to JSON.
+	// If unset, DefaultUrlResolver is used.
+	UrlResolver UrlResolver
+}
+
+// UrlResolver resolves a type URL into an instance of the associated message.
+// This is used for serializing and de-serializing Any messages to and from
+// JSON.
+type UrlResolver interface {
```

Comment:
I'd suggest naming this TypeResolver or AnyResolver to be more specific.  And same for the field names in Marshaler and Unmarshaler.

Reply:
Done. Went with `AnyResolver` since `Any` is the only thing this is used for.